Zend/zend_alloc.h

```diff
@@ -281,6 +282,19 @@ ZEND_API void zend_mm_get_custom_handlers(zend_mm_heap *heap,
                                           void  (**_free)(void*),
                                           void* (**_realloc)(void*, size_t));
 
+typedef struct _zend_mm_observer zend_mm_observer;
+
+// thread local
+ZEND_API bool zend_mm_is_observed(zend_mm_heap *heap);
+ZEND_API zend_mm_observer* zend_mm_observer_register(
+	zend_mm_heap *heap,
+	void (*malloc)(size_t, void * ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC),
+	void (*free)(void * ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC),
+	void (*realloc)(void *, size_t, void * ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC)
```

Comment:
@arnaud-lb did you meant to suggest something?
If not, what do you think of `malloc_observer`, `free_observer`, `realloc_observer`?

Reply:
no, he means `void (*realloc)(void *old_pointer, size_t size, void *new_pointer ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC)` or something like that, otherwise it's not obvious which `void *` parameter has which meaning.

Reply:
Got it, and done :-D